Warnings in building pkgagent on GCC8

<!-- Before filling this issue, please read the wiki (https://github.com/fossology/fossology/wiki)
and search if the bug do not already exists in the issues (https://github.com/fossology/fossology/issues). -->

### Logs

```lisp
pkgagent.c: In function 'ReadHeaderInfo':
pkgagent.c:461:11: warning: 'strncpy' specified bound 128 equals destination size [-Wstringop-truncation]
           strncpy(pi->buildDate,trim(asctime(gmtime((time_t*)tp))),sizeof(pi->buildDate));
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pkgagent.c: In function 'ProcessUpload':
pkgagent.c:378:7: warning: 'strncpy' specified bound 5000 equals destination size [-Wstringop-truncation]
       strncpy(dpi->pFile, PQgetvalue(result, i, 1), sizeof(dpi->pFile));
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pkgagent.c:366:7: warning: 'strncpy' specified bound 5000 equals destination size [-Wstringop-truncation]
       strncpy(dpi->pFile, PQgetvalue(result, i, 1), sizeof(dpi->pFile));
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pkgagent.c:348:7: warning: 'strncpy' specified bound 5000 equals destination size [-Wstringop-truncation]
       strncpy(pi->pFile, PQgetvalue(result, i, 1), sizeof(pi->pFile));
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/pkgagent/agent/pkgagent.c
+++ b/src/pkgagent/agent/pkgagent.c
@@ -345,7 +345,8 @@ int ProcessUpload (long upload_pk)
      * */
     if (!strcasecmp(mimetype,"application/x-rpm")) {
       pi->pFileFk = atoi(PQgetvalue(result, i, 0));
-      strncpy(pi->pFile, PQgetvalue(result, i, 1), sizeof(pi->pFile));
+      strncpy(pi->pFile, PQgetvalue(result, i, 1), sizeof(pi->pFile)-1);
+      pi->pFile[sizeof(pi->pFile)-1] = '\0';
       repFile = fo_RepMkPath("files", pi->pFile);
       if (!repFile) {
         LOG_FATAL("pfile %ld PkgAgent unable to open file %s",
@@ -363,7 +364,8 @@ int ProcessUpload (long upload_pk)
     }
     else if (!strcasecmp(mimetype, "application/x-debian-package")){
       dpi->pFileFk = atoi(PQgetvalue(result, i, 0));
-      strncpy(dpi->pFile, PQgetvalue(result, i, 1), sizeof(dpi->pFile));
+      strncpy(dpi->pFile, PQgetvalue(result, i, 1), sizeof(dpi->pFile)-1);
+      dpi->pFile[sizeof(dpi->pFile)-1] = '\0';
       if (GetMetadataDebBinary(upload_pk, dpi) != -1){
         if (RecordMetadataDEB(dpi) != 0) continue;
       }
@@ -375,7 +377,8 @@ int ProcessUpload (long upload_pk)
     }
     else if (!strcasecmp(mimetype, "application/x-debian-source")){
       dpi->pFileFk = atoi(PQgetvalue(result, i, 0));
-      strncpy(dpi->pFile, PQgetvalue(result, i, 1), sizeof(dpi->pFile));
+      strncpy(dpi->pFile, PQgetvalue(result, i, 1), sizeof(dpi->pFile)-1);
+      dpi->pFile[sizeof(dpi->pFile)-1] = '\0';
       repFile = fo_RepMkPath("files", dpi->pFile);
       if (!repFile) {
         LOG_FATAL("pfile %ld PkgAgent unable to open file %s\n",
@@ -458,7 +461,8 @@ void ReadHeaderInfo(Header header, struct rpmpkginfo *pi)
         case RPMTAG_BUILDTIME:
           t = atol(msgstr);
           tp = &t;
-          strncpy(pi->buildDate,trim(asctime(gmtime((time_t*)tp))),sizeof(pi->buildDate));
+          strncpy(pi->buildDate,trim(asctime(gmtime((time_t*)tp))),sizeof(pi->buildDate)-1);
+          pi->buildDate[sizeof(pi->buildDate)-1] = '\0';
           break;
         case RPMTAG_VENDOR:
           EscapeString(msgstr, pi->vendor, sizeof(pi->vendor));

